backend/src/data_source/material.rs

```diff
@@ -0,0 +1,16 @@
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Material {
+    pub material_id: Uuid,
+    pub name_eng: String,
+    pub name_nl: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct UserMaterial {
+    pub user_id: Uuid,
+    pub material_id: Uuid,
+    pub material_amount: i64,
+}
```

Comment:
In practice, we would (hopefully) never need this table in the Rust code. Instead, we would the SQL ˋjoinˋ syntax to combine them with the users where necessary.

Reply:
@pohlm01 Isn't this struct still required whenever we want to add user material? Like this:
Or maybe I am missing something, I understand that for getting the data we can use SQL 'join' !

`impl MaterialStore {
    pub async fn add_user_material(
        &self,
        user_id: &Uuid,
        material_id: &Uuid,
        material_amount: i32,
    ) -> AppResult<(UserMaterial)> {
        sqlx::query!(
            r#"
            INSERT INTO "user_material" (user_id, material_id, material_amount)
            VALUES ($1, $2, $3)
            ON CONFLICT (user_id, material_id) DO UPDATE
            SET material_amount = $3
            RETURNING
                user_id,
                material_id,
                material_amount
            "#,
            user_id,
            material_id,
            material_amount
        )
            .fetch_one(&self.db)
            .await?
            .try_into()
    }
}`

Reply:
Yes, you are right